src/main/java/net/fabricmc/loader/game/GameProvider.java

```diff
@@ -17,21 +17,39 @@
 package net.fabricmc.loader.game;
 
 import net.fabricmc.api.EnvType;
+import net.fabricmc.loader.api.metadata.ModMetadata;
 import net.fabricmc.loader.entrypoint.EntrypointTransformer;
 
+import java.net.URL;
 import java.nio.file.Path;
+import java.util.Collection;
 import java.util.List;
 
 public interface GameProvider {
 	String getGameId();
 	String getGameName();
+	String getRawGameVersion();
+	String getNormalizedGameVersion();
+	Collection<BuiltinMod> getBuiltinMods();
```

Comment:
They are passive, there shouldn't be any observable load order differences.

Even if that wasn't the case I don't think ordering should play a role anywhere - dependencies and other conflict resolution strategies are much better than trying to control side effects through load ordering. Ordering can only solve a set of conflicts where each part has the same dependency direction, it is ineffective otherwise.